src/components/cesium_map/cesium_UI.js

```diff
@@ -80,7 +100,14 @@ class CesiumMap extends React.Component {
     viewer.addPointPrimitives(setPrimitive);
     viewer.addDataSource(new PointStreamDatasource("BB points")).then((res) => { setPoints = res });
     searchFields = this.props.searchFields;
-    oboePrimitive = setPrimitive.load({ searchFields: this.props.searchFields, rows: 100000 });
+    updatePrimitive(lat, long)
+
+    // set time interval to check the current view every 3 seconds and update points
+    setInterval(() => {
+      if (Math.abs(viewer.currentView.latitude - lat) > 5 || Math.abs(viewer.currentView.longitude - long) > 5) {
+        updatePrimitive(viewer.currentView.latitude, viewer.currentView.longitude)
+      }
+    }, 3000)
```

Comment:
Every 3 seconds, `setInterval` will check if the users move the map and rerender based on the new position.

Reply:
The difference of 5 should probably be adjusted for the zoom level. 5 probably works well for a global view, but not so much when zoomed in quite close.